dev/com.ibm.ws.security.audit.source/src/com/ibm/ws/security/audit/encryption/AuditCrypto.java

```diff
@@ -2109,6 +2113,68 @@ static final byte[] decrypt(byte[] mesg, byte[] key, String cipher, boolean useJ
         return tmpMesg;
     }
 
+    /**
+     * @param key
+     * @param cipher
+     * @return
+     * @throws InvalidKeyException
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeySpecException
+     */
+    private static SecretKey constructSecretKey(byte[] key, String cipher) throws InvalidKeyException, NoSuchAlgorithmException, InvalidKeySpecException, NoSuchProviderException {
+        SecretKey sKey = null;
+        if (cipher.indexOf("AES") != -1) {
+            // 16 bytes = 128 bit key
+            sKey = new SecretKeySpec(key, 0, 16, "AES");
+        } else {
+            DESedeKeySpec kSpec = new DESedeKeySpec(key);
+            SecretKeyFactory kFact = null;
+
+            kFact = (provider == null) ? SecretKeyFactory.getInstance(encryptAlgorithm) : SecretKeyFactory.getInstance(encryptAlgorithm, provider);
```

Comment:
encryptAlgorithm is based off of FIPS being enabled, not on if FIPS 140-3 is enabled.  As such the logic is changing for 140-2.  Is this what you want or do you want 140-2 to behave like it did previously?

Reply:
We want 140-2 behave like it did previously.

Reply:
`provider` being non null happens when using 140-3 or not.  It would seem that the change should be:
```suggestion
            kFact = (!fips140_3Enabled) ? SecretKeyFactory.getInstance(encryptAlgorithm) : SecretKeyFactory.getInstance(encryptAlgorithm, provider);
```
That would give the behavior like it was before to always just pass the encyrptAlgorithm, unless you want the behavior to change to pass a provider in a non 140-3 scenario.

Reply:
This block of code is for previous support. 
For FIPS 140-3 we are using cipher idex of "AES" so we are using the SecretKeySpec() method.

Reply:
I am very confused.  The else block before had `kFact = SecretKeyFactory.getInstance(ENCRYPT_ALGORITHM);`.  `provider` was NOT passed to it.  Now `provider` is passed if it is not null which is possible with and without 140-3 being enabled.  This is why I am concerned because we are doing it different than before.  I am not satisfied with an answer yet as why we changed this behavior.

Reply:
hahaha!!! Now I got it .. you are concern that we pass in the provider now and that can cause a behavior changes. I like to move to the model where we know which provider we want to call. But this is the old code, I will keep it the old way and when we hit the problem in the Semeru ... I can revisit it.